collections_hierarchy/mixins.py

```diff
@@ -1,54 +1,81 @@
+from copy import deepcopy
+
 class ComparableMixin(object):
+    #Implements comparison functionality. Common for List and Dict
     def __eq__(self, other):
-        pass
+        data1 = getattr(self, self.DATA_ATTR_NAME) #Get the value of the property that stores the data
+        data2 = getattr(other, other.DATA_ATTR_NAME)
+        
+        if isinstance(data1, dict):
```

Comment:
You want to be careful here as you are not checking that data2 is of the same type as data1. Perhaps something like:
```
if isinstance(data1, dict) and isinstance(data2, dict)
```